### PR TITLE
remove preferred batch size

### PR DESCRIPTION
**Pull request by debermudez**

### CODE REVIEW

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

File: qa/L0_config_search/test_config_generator.py [hunk 1]
```diff
@@ -127,17 +141,26 @@ def generate_max_limit_with_param_and_model_config(self):
...
-            "run_config_search_preferred_batch_size_disable": True,
             "profile_models": self.profile_models,
         }
         self._write_file(6, 2, 4, 1, model_config)
 
+    def _calculate_total_params(self,
+                                concurrency,
+                                instance_count,
+                                default_config_count=1):
+        concurrency_count = len(range(1, concurrency + 1))
```
[the remark this answers is not in the document]
Comment (debermudez) [reply qa/L0_config_search/test_config_generator.py]: Yea. I updated this function to remove some of the complication. It is also commented now.

File: qa/L0_cpu/test.sh [hunk 2]
```diff
@@ -37,8 +37,13 @@ rm -rf $OUTPUT_MODEL_REPOSITORY
 
 python3 test_config_generator.py --model-names resnet_v1_50_cpu_graphdef
 
-# Compute expected columns (2 instance count * conccurrency * 7 dynamic batch size)
-let "TEST_OUTPUT_NUM_ROWS = 42"
+# Compute expected rows: concurrency * (instance count * dynamic batch size + 1)
+# concurrency:          3       ->  [1, 2, 4]; set in config-profile.yml
+# instance count:       2         ->  [1, 2]; set in config-profile.yml; 
+# dynamic batch size:   1       ->  [1]; now is either on or off
```
[the remark this answers is not in the document]
Comment (debermudez) [reply qa/L0_cpu/test.sh:43]: Updated the comment to reflect that it is always on now.

File: model_analyzer/reports/report_manager.py [hunk 3]
```diff
@@ -396,7 +396,7 @@ def _get_dynamic_batching_phrase(self, config):
         elif dynamic_batching_str == "Enabled":
             dynamic_batch_phrase = "dynamic batching enabled"
         else:
-            dynamic_batch_phrase = f"preferred batch size of {dynamic_batching_str}"
+            dynamic_batch_phrase = f"dynamic batching unknown"
```
[the remark this answers is not in the document]
Comment (debermudez) [reply model_analyzer/reports/report_manager.py:399]: Added an assertion. The unknown case should not happen so the assertion should guard us against an issue now.

File: model_analyzer/config/run/run_search.py [hunk 4]
```diff
@@ -250,29 +230,18 @@ def _sweep_model_config_only(self, model):
...
-            if dynamic_batching_enabled:
-                batch_size_limit_reached = self._model_config_parameters[
-                    'dynamic_batching'] > self._max_preferred_batch_size
-                if batch_size_limit_reached:
-                    return model, []
+            return model, []
         return model, [self._create_model_config(cpu_only=model.cpu_only())]
 
+    def _enable_dynamic_batching(self):
+        self._model_config_parameters['dynamic_batching'] = None
```
[the remark this answers is not in the document]
Comment (debermudez) [reply model_analyzer/config/run/run_search.py:243]: Good call. Added a comment.

File: docs/config_search.md [hunk 5]
```diff
@@ -51,9 +51,7 @@ For
 Model Analyzer tries values from 1 to 5. This value can be changed using the
 `run_config_search_max_instance_count` key in the Model Analyzer Config. For
 [`dynamic_batching`](https://github.com/triton-inference-server/server/blob/master/docs/model_configuration.md#dynamic-batcher)
-settings, Model Analyzer tries enabling/disabling dynamic batching plus values
-from 1 to 16 for the `preferred_batch_size`. This value can be changed using the
-`run_config_search_max_preferred_batch_size` key in the Model Analyzer Config.
+settings, Model Analyzer tries enabling/disabling dynamic batching.
```
[the remark this answers is not in the document]
Comment (debermudez) [reply docs/config_search.md:54]: Corrected.

File: qa/L0_config_search/test_config_generator.py [hunk 6]
```diff
@@ -127,16 +141,30 @@ def generate_max_limit_with_param_and_model_config(self):
...
+        total_param_count = self._calculate_total_params(
+            concurrency_count, instance_count)
+        self._write_file(total_param_count, 2, 4, 1, model_config)
+
+    def _calculate_total_params(self,
+                                concurrency_count,
+                                instance_count,
+                                default_config_count=1):
+        """
+        Calculate the total number of configs that MA will try
```
Comment (tgerdesnv) [inline qa/L0_config_search/test_config_generator.py:165]: Some more details of the args would be nice:

Given the count of concurrencies and instances to sweep over, calculate and return the total number of parameter combinations that MA will try.  default_config_count indicates how many extra model configs will be run in addition to the normal sweep, and should be set to 0 if the default config is already part of the sweep space.

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**